### PR TITLE
Add Proxy context and subscribe to proxies

### PR DESCRIPTION
**Pull request by TingALin**
Addresses #413 
https://forum.polkadot.network/t/proxies-to-have-delegate-proxies-reverse-lookup/2312 in detail

- [x] Initialises `Accounts/Proxies` context.
- [x] Subscribes to the imported accounts proxies.
- [x] Re-organises the discovered proxies into `delegate -> [proxiedAccounts]`.
- [x] Adds a `proxiedAccounts` function to fetch delegators of a particular delegate.

### CODE REVIEW

**Comment by TingALin** [comment]
Regarding the next step, thinking about one of the following functions:

- isDelegate(delegatee: MaybeAccount)->boolean
- proxiedAccounts(delegatee: MaybeAccount)-> Array<ProxyAccount>
- proxiedAccounts(delegatee: MaybeAccount)-> Array<MaybeAccount>

File: src/contexts/Accounts/Proxies/index.tsx [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+  const { accounts, activeAccount } = useConnect();
+
+  // store the proxy accounts of each imported account.
+  const [proxies, setProxies] = useState<Array<Proxy>>([]);
+  const proxiesRef = useRef(proxies);
+
+  const [unsubs, setUnsubs] = useState<AnyApi>([]);
+  const unsubsRef = useRef(unsubs);
+
+  const [delegators, setDelegators] = useState<Array<MaybeAccount>>([]);
```
Comment (rossbulat) [inline src/contexts/Accounts/Proxies/index.tsx:40]: Why is this `delegators`? We need to store `delegates`, and the above `proxies` state stores the delegators, no?

This should not be `MaybeAccount` as there will never be a `null` value being stored. In addition, we need to store more than the address here. We also need the "real" account, or delegator account alongside it.

File: src/contexts/Accounts/Proxies/index.tsx [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+  const delegatorsRef = useRef(delegators);
+
+  useEffect(() => {
+    if (isReady) {
+      // local updated values
+      let newProxies = proxiesRef.current;
```
Comment (rossbulat) [inline src/contexts/Accounts/Proxies/index.tsx:46]: this line is not being used, just do `const newProxies` on line 61.

File: src/contexts/Accounts/Proxies/index.tsx [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+  useEffect(() => {
+    Object.values(unsubsRef.current).forEach(({ unsub }: AnyApi) => {
+      unsub();
+    });
+  }, []);
+
+  useEffect(() => {
+    const allProxies = [];
+    const proxiedAccounts = [];
+    for (const a of accounts) {
```
Comment (rossbulat) [inline src/contexts/Accounts/Proxies/index.tsx:98]: Why is `accounts` being used? This `useEffect` just needs to loop through `proxiesRef.current` and reverse the record to be `delegate -> proxies` instead.

File: src/contexts/Accounts/Proxies/index.tsx [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+    setStateWithRef(
+      unsubsRef.current.concat({ key: address, unsub }),
+      setUnsubs,
+      unsubsRef
+    );
+    return unsub;
+  };
+
+  // Gets the proxy accounts for a given proxy via the delegate
+  const getProxyAccounts = (address: MaybeAccount) => {
```
Comment (rossbulat) [inline src/contexts/Accounts/Proxies/index.tsx:165]: Should this be renamed to `getDelegates`?

**Review by rossbulat — CHANGES_REQUESTED** [review]

**Review by rossbulat — APPROVED** [review]

**Pull request merged**